src/libraries/System.Collections/tests/Generic/Comparers/Comparer.Generic.Tests.cs

```diff
@@ -78,7 +78,7 @@ public void Comparer_IComparerCompareWithObjectsNotOfMatchingTypeShouldThrow()
             // throw if both inputs are non-null and one of them is not of type T
             IComparer comparer = Comparer<T>.Default;
             StrongBox<T> notOfTypeT = new StrongBox<T>(default(T));
-            if (default(T) != null) // if default(T) is null these asserts will fail as IComparer.Compare returns early if either side is null
+            if (typeof(T).IsValueType) // if default(T) is null these asserts will fail as IComparer.Compare returns early if either side is null
```

Comment:
This is a test and doesn't need to be changed.  (The comment would also be stale if it were changed.)